packages/date-picker/src/FilterDateRangePicker/components/DateRangeInputField/DateRangeInputField.tsx

```diff
@@ -0,0 +1,129 @@
+  /**
+   * A description that provides context for the text field
+   */
+  description?: React.ReactNode
+  isReversed?: boolean
+  /**
+   * Updates the styling of the validation FieldMessage
+   */
+  status?: FieldMessageStatus
+  /**
+   * A descriptive message for `status` states
+   */
+  validationMessage?: React.ReactNode
+  disabled?: boolean
+}
+
+export type DateRangeInputFieldRefs = {
+  inputRangeStartRef?: React.RefObject<HTMLInputElement>
+  inputRangeEndRef?: React.RefObject<HTMLInputElement>
+}
+
+export const DateRangeInputField = React.forwardRef<
+  DateRangeInputFieldRefs,
+  DateRangeInputFieldProps
+>(
+  (
+    {
+      id,
+      legend,
+      inputRangeStartProps,
+      inputRangeEndProps,
+      description,
+      disabled,
+      isReversed = false,
+      status,
+      validationMessage,
+      locale,
+      ...fieldGroupProps
+    },
+    ref
+  ) => {
+    const customRefObject = isRefObject(ref) ? ref.current : null
+    const inputRangeStartRef = customRefObject?.inputRangeStartRef
+    const inputRangeEndRef = customRefObject?.inputRangeEndRef
+
+    const descriptionId = `${id}--field-message`
+    const errorMessageId = `${id}--error-message`
+
+    const shouldShowValidationMessage = !disabled && validationMessage
+
+    const inputDescribedBy = shouldShowValidationMessage
+      ? `${errorMessageId} ${descriptionId}`
+      : descriptionId
+
+    return (
+      <FieldGroup {...fieldGroupProps}>
+        <fieldset className={styles.dateRangeInputContainer}>
+          <legend>
+            <VisuallyHidden>{legend}</VisuallyHidden>
+          </legend>
+          <DateInput
+            ref={inputRangeStartRef}
+            id={`${id}--from`}
+            aria-describedby={inputDescribedBy}
+            autoComplete="off"
+            disabled={disabled}
+            status={status}
+            {...inputRangeStartProps}
+            classNameOverride={classnames(
+              styles.inputRangeStart,
+              inputRangeStartProps.classNameOverride
+            )}
+          />
+          <DateInput
+            ref={inputRangeEndRef}
+            id={`${id}--to`}
+            aria-describedby={inputDescribedBy}
+            autoComplete="off"
+            disabled={disabled}
+            status={status}
+            {...inputRangeEndProps}
+            classNameOverride={classnames(
+              styles.inputRangeEnd,
+              inputRangeEndProps.classNameOverride
+            )}
+          />
+        </fieldset>
+
+        {shouldShowValidationMessage && (
+          <FieldMessage
+            id={errorMessageId}
+            message={validationMessage}
+            status={status}
+            reversed={isReversed}
+          />
+        )}
+        <FieldMessage
+          id={descriptionId}
+          message={formatInputDescription(description, locale)}
+          reversed={isReversed}
+          classNameOverride={disabled ? styles.disabled : undefined}
+        />
```

Comment:
Shouldn't descriptions come before validation messages?

Reply:
I don't believe that's how our components are? 🤔

Reply:
Oops no you're right, this PR was wrong haha https://github.com/cultureamp/kaizen-design-system/pull/3169